### PR TITLE
Add definition of IStream

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This is needed for implementing `IMetaDataEmit::SaveToStream` on non-Windows.

### CODE REVIEW

File: src/inc/dncp.h [hunk 1]
```diff
@@ -93,6 +93,24 @@
...
+
+    typedef union
+    {
+        struct
+        {
+            DWORD LowPart;
+            DWORD HighPart;
+        } u;
+        ULONGLONG QuadPart;
+    } ULARGE_INTEGER;
```
Comment (AaronRobinsonMSFT) [inline src/inc/dncp.h:113]:
```suggestion
    typedef union {
        struct {
            DWORD LowPart;
            DWORD HighPart;
        } u;
        ULONGLONG QuadPart;
    } ULARGE_INTEGER;
```

File: src/inc/dncp.h [hunk 2]
```diff
@@ -212,8 +233,9 @@ HRESULT PAL_IIDFromString(LPCOLESTR, IID*);
         #include <unknwn.h>
 
         // Unusable COM and RPC types
+        struct tagSTATSTG;
+        typedef struct tagSTATSTG STATSTG;
```
Comment (AaronRobinsonMSFT) [inline src/inc/dncp.h:237]:
```suggestion
        using STATSTG = struct tagSTATSTG;
```

Comment (AaronRobinsonMSFT) [inline src/inc/dncp.h:237]:
```suggestion
        using STATSTG = tagSTATSTG;
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
@AaronRobinsonMSFT this is ready to be merged (I don't have permissions in this repo)

**Pull request merged**